mockkubeapiserver/patchresource.go

```diff
@@ -81,20 +85,50 @@ func (req *patchResource) Run(ctx context.Context, s *MockKubeAPIServer) error {
 		return req.writeResponse(patched)
 	}
 
+	var updated *unstructured.Unstructured
+	changed := true
 	if req.SubResource == "" {
-		if err := applyPatch(existingObj.Object, body.Object); err != nil {
-			klog.Warningf("error from patch: %v", err)
-			return err
+		if resource.TypeInfo != nil {
+
+			patchOptions := metav1.PatchOptions{}
+			if fieldManager := req.r.URL.Query().Get("fieldManager"); fieldManager != "" {
+				patchOptions.FieldManager = fieldManager
+			}
+			if force := req.r.URL.Query().Get("force"); force != "" {
+				forceBool, err := strconv.ParseBool(force)
+				if err != nil {
+					return fmt.Errorf("invalid value %q for force", force)
+				}
+				patchOptions.Force = &forceBool
+			}
+			updated, changed, err = resource.DoServerSideApply(ctx, existingObj, bodyBytes, patchOptions)
+			if err != nil {
+				klog.Warningf("error from DoServerSideApply: %v", err)
+				return err
+			}
+
+		} else {
+			klog.Warningf("falling back to untyped apply emulation")
```

Comment:
I think we have a schema for things that don't have a schema, but that still uses ssa. ...